src/solvers/unsteady_solver.C

```diff
@@ -197,6 +197,11 @@ void UnsteadySolver::adjoint_advance_timestep ()
       // The adjoint system has been solved. We need to store the adjoint solution and
       // load the primal solutions for the next time instance (t - delta_ti).
       _system.time -= _system.deltat;
+
+      if(_system.time < 0.0)
+      {
+        libmesh_error_msg("Error ! Time has become negative.");
```

Comment:
Should this be a warning?  We typically start simulating at t=0, but that's a habit, not an intended requirement.  I could imagine someone simulating something (rocket startup?) where t=0 has a different mid-simulation meaning.  And since we're stepping back and forth by deltat here, don't we expect at least some numeric drift to be possible, even likely?

Reply:
Agreed, will make this into a warning, and change the 0 to a f(TOLERANCE).